main.go

```diff
@@ -181,6 +183,24 @@ func parseAndRun(out output) {
 	runWithOptions(*opts, out)
 }
 
+// overrideDefaults clears fields in the default options that may
+// clash with user-specified options.
+// E.g., if the defaults has a peer list file, and the user has specifed
+// a peer through the command line, then the final options should only
+// contain the peer specified in the args.
+func overrideDefaults(defaults *Options, args []string) {
+	argsParser, argsOnly := newParser()
+	argsParser.ParseArgs(args)
+
+	// Clear default peers if the user has specified peer options in args.
+	if len(argsOnly.TOpts.HostPorts) > 0 && defaults.TOpts.HostPortFile != "" {
+		defaults.TOpts.HostPortFile = ""
```

Comment:
Do you really need the second check? If you always set `HostPortFile` to `""` when `HostPorts` is non-empty, you get the same result.